src/main/java/com/gdsc/blog/security/MyUserDetails.java

```diff
@@ -1,39 +1,38 @@
 package com.gdsc.blog.security;
 
 
+import com.gdsc.blog.user.entity.User;
+import com.gdsc.blog.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
-import com.gdsc.blog.user.entity.User;
-import com.gdsc.blog.user.repository.UserRepository;
-import lombok.RequiredArgsConstructor;
-
 
 
 @Service
 @RequiredArgsConstructor
 public class MyUserDetails implements UserDetailsService {
 
-  private final UserRepository userRepository;
-
-  @Override
-  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-    final User user = userRepository.findByUsername(username).get();
-
-    if (user == null) {
-      throw new UsernameNotFoundException("User '" + username + "' not found");
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        final User user = userRepository.findByUsername(username).get();
+
+        if (user == null) {
+            throw new UsernameNotFoundException("User '" + username + "' not found");
+        }
```

Comment:
이 부분에서, `optional.get()`은 데이터를 반환하고 없거나 null일 경우 `NoSuchElement`를 throw하기 때문에, 아래의 if문이 절대로 동작하지 않게 됩니다.

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--

따라서, 의도하고자 하는 동작을 위해서는 `.orElse(null)` 혹은 `.orElseThrow()` 메소드를 사용해야 합니다.
```suggestion
        // Example 1(추천 안함)
        final User user = userRepository.findByUsername(username).orElse(null);
        if (user == null) {
            throw new UsernameNotFoundException("User '" + username + "' not found");
        }
        // Example 2
        final User user = userRepository.findByUsername(username).orElseThrow(() ->
            new UsernameNotFoundException("User '" + username + "' not found"));
       
```